pkg/dns/aws/dns.go

```diff
@@ -174,6 +178,38 @@ func (m *Manager) getZoneID(zoneConfig configv1.DNSZone) (string, error) {
 	return id, nil
 }
 
+// getLBHostedZone finds the hosted zone ID of an ELB whose DNS name matches the
+// name parameter. Results are cached.
+func (m *Manager) getLBHostedZone(name string) (string, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	if id, exists := m.lbZones[name]; exists {
+		return id, nil
+	}
+
+	var id string
+	loadBalancers, err := m.elb.DescribeLoadBalancers(&elb.DescribeLoadBalancersInput{})
```

Comment:
`DescribeLoadBalancers` call seems to be paginated
https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_DescribeLoadBalancers.html

Reply:
Don’t have time to fix it right now and this shouldn’t be any worse than before — will leave it to others to decide what they wanna do before Monday

Reply:
Fixed